bftengine/src/preprocessor/messages/PreProcessReplyMsg.cpp

```diff
@@ -87,6 +87,15 @@ void PreProcessReplyMsg::validate(const ReplicasInfo& repInfo) const {
   }
 }  // namespace preprocessor
 
+std::vector<char> PreProcessReplyMsg::getResultHashSignature() const {
```

Comment:
nit: [very minor comment] void PreProcessReplyMsg::getResultHashSignature(std::vector<char> & result_hash) -- This will avoid result vector and copy ctors.

Reply:
Actually no copy happens here too. RVO is applied for return-by-value vector. 

This function is called in RequestProcessingState::handlePreProcessReplyMsg(), as a parameter for emplace_back() which accepts universal reference. This is one of the cases where copy elision is applied so we are fine I think.

Reply:
yeah, i missed the RVO. this is absolutely fine. Thanks @tdimitrov